homeassistant/components/voice_assistant/pipeline.py

```diff
@@ -25,18 +26,25 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+STORAGE_KEY = f"{DOMAIN}.pipelines"
+STORAGE_VERSION = 1
+
+SAVE_DELAY = 10
+
 
 @callback
 def async_get_pipeline(
     hass: HomeAssistant, pipeline_id: str | None = None, language: str | None = None
 ) -> Pipeline | None:
     """Get a pipeline by id or create one for a language."""
+    pipeline_store: PipelineStore = hass.data[DOMAIN]
+
     if pipeline_id is not None:
-        return hass.data[DOMAIN].get(pipeline_id)
+        return pipeline_store.pipelines.get(pipeline_id)
 
     # Construct a pipeline for the required/configured language
     language = language or hass.config.language
-    return Pipeline(
+    return pipeline_store.async_add(
```

Comment:
I thought we were going to drop this? Or is that for a next one.

Reply:
Yeah, I think that should be in a follow-up 👍